floris/tools/optimization/scipy/yaw.py

```diff
@@ -171,24 +198,71 @@ def _optimize(self):
         Returns:
             opt_yaw_angles (np.array): optimal yaw angles of each turbine.
         """
+        opt_yaw_angles = np.array(self.x0, dtype=float)
```

Comment:
I feel like it makes more sense for the opt_yaw_angle values for turbines that are not optimized to be the baseline yaw angles as opposed to x0. Let's say for example someone just sets x0 to 15 degrees for all turbines, then some downstream turbines are removed from the set of control variables. I think it would be more appropriate to set the yaw angles for the uncontrolled turbines to their baseline values in this case.

Reply:
I totally see your point. I was struggling with this a little bit. The problem is that the user will always define an `x0` for all turbines in the wind farm, even for downstream turbines. If those downstream turbines are excluded, then what value do we fix those turbines to? It could either be `x0`, since that would be the initial condition for the optimization, or it can be the default value inside the `floris` object. I think there is no wrong or right and I can totally foresee confusion with the user here. Here's my idea now: I'll follow your suggestion to use the `self.x_baseline` values as the fixed values for downstream turbines, and so `x0` is only used as an initial guess in the optimization for turbines which are having their yaw angles actually optimized. Though, I will raise a warning to the user if `self.x_baseline` contains any non-zero values, to make sure that a future user will not accidentally fix a downstream turbine's yaw angle to a nonzero value.

Reply:
I was struggling with this too, even after I posted the comment. I see your point that there is no right or wrong answer, and perhaps I was being too nit-picky (since this is kind of an edge case). How about you go with what you think makes the most sense here. But what's probably most important is to just add to the documentation what values will be used for uncontrolled turbines so it's clear to users. And that sounds like a good idea to add a warning if it would be setting uncontrolled turbines to non-zero values.

Reply:
Okay I did some more thinking and decided it would be best to simply add an additional input to the optimization class defining the baseline yaw angles. This automatically makes a clear separation between `yaw_angles_baseline` and `x0`. I have updated the documentation accordingly such that users should not get confused. `x0` is now only used as an initial guess in the optimization and nowhere else. Actually, only a subset of `x0` is used if `exclude_downstream_turbines=True` or if any turbine's yaw angle is fixed through an equality constraint using `bnds`, since then `self.turbs_to_opt` is shorter than the total number of turbines in the farm.

If `yaw_angles_baseline=None` and thus is unspecified, then the values are derived from the `floris` object itself, like @ejsimley suggested. To avoid accidentally inserting wrong baseline values, I have added a `print` statement in case any of those values are not `0.0`:
```
    if any(np.abs(self.yaw_angles_baseline) > 0.0):
        print("INFO: Baseline yaw angles were not specified and were derived from the floris object.")
        print("INFO: The inherent yaw angles in the floris object are not all 0.0 degrees.")
```

and also I inform the user if `self.x0` differs from `self.yaw_angles_baseline`, since I do not expect this to be a common case and want users to make sure they are doing what they intent to do:
```
if not all(np.array(x0) == np.array(self.yaw_angles_baseline)):
      print("INFO: Baseline yaw angles in FLORIS differ from initial guess in optimization (self.x0)")
```